pmd-visualforce/etc/grammar/VfParser.jjt

```diff
@@ -255,8 +260,9 @@ PARSER_END(VfParser)
 
 <HtmlScriptContentState> TOKEN :
 {
-		 <HTML_SCRIPT_CONTENT: (~[]) >
-		| <HTML_SCRIPT_END_TAG : "</script>" > : AfterTagState
+		  <HTML_SCRIPT_END_TAG : "</script>" > : AfterTagState
+		| <EL_EXPRESSION_IN_SCRIPT: "{!" (<WHITESPACES>)? > : ElInScriptState
```

Comment:
@jsotuyod this is a parser for VF, not Javascript. :) 
VF is a very opinionated platform, which treats {! } construct anywhere in code 100% of the time as EL; therefore, any javascript construct that looks like {!varName} would result in compilation error because there would be a missing property with that name.